src/zenml/integrations/kubeflow/orchestrators/local_deployment_utils.py

```diff
@@ -203,21 +207,53 @@ def start_kfp_ui_daemon(pid_file_path: str, port: int) -> None:
         f"{port}:80",
     ]
 
-    def _daemon_function() -> None:
-        """Port-forwards the Kubeflow Pipelines UI pod."""
-        subprocess.check_call(command)
-
-    if sys.platform == "win32":
+    if not port_available(port):
+        modified_command = command.copy()
+        modified_command[-1] = "PORT:80"
+        logger.warning(
+            "Unable to port-forward Kubeflow Pipelines UI to local port %d "
+            "because the port is occupied. In order to access the Kubeflow "
+            "Pipelines UI at http://localhost:PORT/, please run '%s' in a "
+            "separate command line shell (replace PORT with a free port of "
+            "your choice).",
+            port,
+            " ".join(modified_command),
+        )
+    elif sys.platform == "win32":
         logger.warning(
-            f"Daemon functionality not supported on Windows. "
-            f"In order to access the Kubeflow Pipelines UI, please run "
-            f"'{' '.join(command)}' in a separate command line shell."
+            "Daemon functionality not supported on Windows. "
+            "In order to access the Kubeflow Pipelines UI at "
+            "http://localhost:%d/, please run '%s' in a separate command "
+            "line shell.",
+            port,
+            " ".join(command),
         )
     else:
         from zenml.utils import daemon
 
+        def _daemon_function() -> None:
+            """Port-forwards the Kubeflow Pipelines UI pod."""
+            subprocess.check_call(command)
+
         daemon.run_as_daemon(
-            _daemon_function,
-            pid_file=pid_file_path,
+            _daemon_function, pid_file=pid_file_path, log_file=log_file_path
+        )
+        logger.info(
+            "Started Kubeflow Pipelines UI daemon (check the daemon logs at %s "
```

Comment:
Can we add a simple way to hook into this file? Like `zenml orchestrator logs kubeflow_orchestrator` that would basically do a `tail -f {pid_file_path}`

Reply:
There already is `zenml orchestrator monitor` which does exactly this, do you think we should forward use this functionality for these port-forwarding logs? To me it feels like this is not a really interesting/relevant log file that is worth monitoring.

Reply:
What does `zenml orchestrator monitor` do? I didnt know we had it. If its already there then lets just leave it like that!

Reply:
Basically a `tail -f`, but only for one specific log file per orchestrator at the moment. Should I point it to this log file for kubeflow?

Reply:
Yes I think it makes sense right? Or is there another file to monitor? It would also be nice to print that command as a log after `zenml stack up` finishes

Reply:
We could think about maybe forwarding it to some kubeflow pod logs instead, but we can start using this file for now.
I'll also mention it on `zenml stack up` as a log